src/jit/emitxarch.cpp

```diff
@@ -10832,6 +10832,9 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
 
             dst += emitOutputByte(dst, emitGetInsSC(id));
             sz = emitSizeOfInsDsc(id);
+
+            // Kill any GC ref in the destination register.
+            emitGCregDeadUpd(id->idReg1(), dst);
```

Comment:
On second further thinking, we can avoid call overhead for those instructions that write to an sse/avx reg by conditioning this on emitInsCanOnlyWriteSSE2OrAVXReg()?

e.g. if (emitInsCanOnlyWriteSSE2OrAVXReg())
       {
           // do nothing
       } 
      else
       {
             emitGCregDeadUpd(id->idReg1(), dst);
      }

I see emitInsCanOnlyWriteSSE2OrAVXReg() needs to be updated to handle pextrw and pmovkskb instructions.